packages/compiler/test/teardown/parse.ts

```diff
@@ -0,0 +1,15 @@
+import { test } from 'uvu';
+import * as assert from 'uvu/assert';
+import { parse, teardown } from '@astrojs/compiler';
+
+const FIXTURE = `<div>hello</div>`;
+
+test('parse still works after teardown', async () => {
+  const ast1 = await parse(FIXTURE);
+  assert.ok(ast1);
+  teardown();
+  const ast2 = await parse(FIXTURE);
+  assert.ok(ast2);
```

Comment:
It'll only re-instantiate if `teardown()` is called once before, but otherwise it would always reuse the one previously instantiated, so calling `parse` consecutively would all share one instance.

I think for Astro, if there isn't a user runtime dependency that uses the Astro compiler, the entire build cycle should all share one instance before the build finishes and it calls `teardown()`

Reply:
Makes sense to me! Thanks for clarifying. I wasn't sure if the intended usage was to call `teardown` after every `transform` call or if we'd only call it once.